File: 948-sort-an-array/sort-an-array.java
class Solution {
    public int[] sortArray(int[] nums) {
        mergeSort(nums , 0 , nums.length -1 );

        return nums;
    }
        //MERGE SORT 

        //CONQUER

        void merge(int arr[] , int L , int M , int R)
        {
            int lsize = M - L + 1;
            int rsize = R - M;

            int left[] = new int[lsize];
            int right[] = new int[rsize];

            for(int i = 0 ; i < lsize ; i++)
            {
                left[i] = arr[L + i];
            }

            for(int j = 0 ; j < rsize ; j++)
            {
                right[j] = arr[M + 1 + j];
            } 

            // POINTERS

            int i = 0 ;
            int j = 0 ;
            int k = L ;

            while( i < left.length && j < right.length)
            {
                if(left[i] <= right[j])
                {
                    arr[k] = left[i];
                    i += 1;
                }
                else
                {
                    arr[k] = right[j];
                    j += 1;
                }
                k += 1;
            }

            while( i < left.length)
            {
                arr[k] = left[i];
                i += 1;
                k += 1;
            }

            while( j < right.length)
            {
                arr[k] = right[j];
                j += 1;
                k += 1;
            }
        }
 
        //DIVIDE 

        void mergeSort(int arr[] , int l , int r)
        {
            if(l >= r)
            {
                return;
            }
            int m = (r + l) / 2 ;

            mergeSort(arr, l , m) ;
            mergeSort(arr , m + 1 , r);

            merge(arr , l , m , r) ;

        }

      

        
        
    }

File: 948-sort-an-array/sort-an-array.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        #MERGE SORT 

        # CONQUER

        def merge(arr , m , l , r):
            leftArr = arr[l : m + 1]
            rightArr = arr[m + 1 : r + 1]
            i = 0 
            j = 0 
            k = l 

            while i < len(leftArr) and j < len(rightArr):

                if leftArr[i] < rightArr[j]:
                    arr[k] = leftArr[i]
                    i += 1
                else:
                    arr[k] = rightArr[j]
                    j += 1
                
                k += 1
            while i < len(leftArr):
                arr[k] = leftArr[i]
                i += 1
                k += 1
            while j < len(rightArr):
                arr[k] = rightArr[j]
                j += 1
                k += 1

            return arr
            




        #DIVIDE 

        def mergeSort(arr , l , r ):

            if l >= r:
                return 

            m = (l + r) // 2

            mergeSort(arr , l , m)
            mergeSort(arr , m + 1 , r)
            merge(arr , m , l , r )


        mergeSort(nums , 0 , len(nums) - 1)

        return nums